datahub-web-react/src/app/ingestV2/ManageIngestionPage.tsx

```diff
@@ -78,9 +99,64 @@ export const ManageIngestionPage = () => {
     const [showCreateSecretModal, setShowCreateSecretModal] = useState<boolean>(false);
     const [hideSystemSources, setHideSystemSources] = useState(true);
 
+    const [capabilitySummary, setCapabilitySummary] = useState<CapabilitySummary | null>(null);
+    const [isCapabilitySummaryLoading, setIsCapabilitySummaryLoading] = useState<boolean>(true);
+    const [isCapabilitySummaryError, setIsCapabilitySummaryError] = useState<string | null>(null);
+
     const history = useHistory();
     const shouldPreserveParams = useRef(false);
 
+    useEffect(() => {
+        const fetchCapabilitySummary = async () => {
+            setIsCapabilitySummaryLoading(true);
+            try {
+                const response = await fetch('/assets/ingestion/capability_summary.json');
+                if (!response.ok) {
+                    throw new Error(`Failed to fetch capability summary: ${response.status} ${response.statusText}`);
+                }
+                const data = await response.json();
+                setCapabilitySummary(data);
```

Comment:
is there a way we can first enforce that the data conforms to the CapabilitySummary type? as it's written, if the json that comes back happens to be non-conforming, we are setting ourselves up for a null pointer exception